### PR TITLE
comment all kustomize patches ...

### PR DESCRIPTION
**Pull request by bavarianbidi**
... to describe the reasons more detailed

towards: https://github.com/giantswarm/roadmap/issues/2440

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,31 @@
...
+
+Following notable commands/scripts are triggered in `make generate`:
+
+1. [`hack/generate-kustomize-patches.sh`](hack/generate-kustomize-patches.sh) 
+    1. fetch the latest upstream released manifest.
+    1. as `JSON Patch 6902` doesn't support `wildcard` patches, we iterate over all `MutatingWebhookConfiguration` and `ValidatingWebhookConfiguration` to create valid `kustomize` patches for all items in the list.
+1. `make delete-generated-helm-charts` to cleanup folder `helm/cluster-api/templates`.
+1. `kustomize build config/helm -o helm/cluster-api/templates` apply all the defined `kustomize` changes (defined in `kustomization.yaml`)
+1. [`hack/move-generated-crds.sh`](hack/move-generated-crds.sh) used to move all the `CRDs` into a own directory as `helm` isn't that good when CRDs needs to get installed.
+1. [`hack/generate-crd-version-patches.sh`](hack/generate-crd-version-patches.sh) --> `¯\_(ツ)_/¯`
```
Comment (bavarianbidi) [inline README.md:22]: @nprokopic (from [this slack thread](https://gigantic.slack.com/archives/C0559SH3RJ4/p1687868292279069)) ... i'm really thinking of dropping this kustomize steps as i didn't get the benefit

File: config/helm/deployment-args-controller-manager.yaml [hunk 2]
```diff
@@ -13,4 +13,3 @@ spec:
             - --feature-gates=MachinePool=true,ClusterResourceSet=true,ClusterTopology=true
             - --watch-filter={{ .Values.watchFilter }}
             - --v=0
-          image: '{{.Values.images.domain}}/{{.Values.images.core.name}}:{{.Values.images.core.tag | default .Values.images.tag}}'
```
Comment (bavarianbidi) [inline config/helm/deployment-args-controller-manager.yaml:16]: we don't need this line as there is already an `image` section in `kustomize` which we can use for the `image: ` value replacement (see `kustomize.yaml` changes in this PR)

File: config/helm/deployment-args-kubeadm-bootstrap-controller-manager.yaml [hunk 3]
```diff
@@ -13,4 +13,3 @@ spec:
             - --feature-gates=MachinePool=true,KubeadmBootstrapFormatIgnition=true
             - --watch-filter={{ .Values.watchFilter }}
             - --v=0
-          image: '{{.Values.images.domain}}/{{.Values.images.bootstrap.name}}:{{.Values.images.bootstrap.tag | default .Values.images.tag}}'
```
Comment (bavarianbidi) [inline config/helm/deployment-args-kubeadm-bootstrap-controller-manager.yaml:16]: we don't need this line as there is already an `image` section in `kustomize` which we can use for the `image: ` value replacement (see `kustomize.yaml` changes in this PR)

File: config/helm/deployment-args-kubeadm-control-plane-controller-manager.yaml [hunk 4]
```diff
@@ -13,4 +13,3 @@ spec:
             - --feature-gates=ClusterTopology=true,KubeadmBootstrapFormatIgnition=true
             - --watch-filter={{ .Values.watchFilter }}
             - --v=0
-          image: '{{.Values.images.domain}}/{{.Values.images.controlplane.name}}:{{.Values.images.controlplane.tag | default .Values.images.tag}}'
```
Comment (bavarianbidi) [inline config/helm/deployment-args-kubeadm-control-plane-controller-manager.yaml:16]: we don't need this line as there is already an `image` section in `kustomize` which we can use for the `image: ` value replacement (see `kustomize.yaml` changes in this PR)

File: README.md (hunk 1 above)
Comment (bavarianbidi) [reply README.md]: updated documentation - resolved for now

**Review by njuettner — APPROVED** [review]

**Review by nprokopic — APPROVED** [review]

**Pull request merged**